src/vm/database/structures.rs

```diff
@@ -218,11 +309,9 @@ impl<'db, 'conn> STXBalanceSnapshot<'db, 'conn> {
             .get(&recipient_key)
             .unwrap_or(STXBalance::zero());
 
-        recipient_balance.amount_unlocked =
-            recipient_balance
-                .amount_unlocked
-                .checked_add(amount)
-                .ok_or(Error::Runtime(RuntimeErrorType::ArithmeticOverflow, None))?;
+        recipient_balance
+            .checked_add_unlocked_amount(amount)
+            .ok_or(Error::Runtime(RuntimeErrorType::ArithmeticOverflow, None))?;
```

Comment:
Is this even an error we want to recover from?  If there are this many STX piled into an address (which shouldn't be possible for thousands of years), then we probably have a show-stopping problem.

Reply:
This is preserving the existing behavior on unlock. This will just make the transaction invalid. We could change this to a panic, but that would alter the current behavior for this case.

Reply:
Ah, right. This is fine then.